### ISSUE TITLE
UserWarning: resource_tracker: There appear to be 11 leaked semaphore objects to clean up at shutdown

### ISSUE THREAD

**Issue by fcoppey** [awslabs/autogluon#2009]
Hello,

I’m using autogluon on Mac OS Monterrey 12.5 with a M1 MacBook Pro,

until recently all was working fine. This morning I updated a few things through brew and now I get this error every time I try to predict from an previously created autogluon predictor. 

Have any clue what could the issue? it happened the same to my colleague recently and now we’re unable to work on our project...

I use autogluon 0.3.1, python 3.8.12

**here is the error as showed in Pycharm**
```
/Users/florentin/.pyenv/versions/3.8.12/lib/python3.8/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 11 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '

Process finished with exit code 245
```

**here is the brew versions of the packages installed:**
atk 2.38.0
autoconf 2.71
azure-cli 2.39.0
ca-certificates 2022-07-19_1
cairo 1.16.0_5
fontconfig 2.14.0 2.13.1
freetype 2.10.4 2.12.1 2.11.1
fribidi 1.0.10 1.0.11 1.0.12
gdbm 1.23
gdk-pixbuf 2.42.8_1
geckodriver 0.31.0
gettext 0.21
git-lfs 3.2.0
glib 2.72.3
gobject-introspection 1.72.0
graphite2 1.3.14
gtk+ 2.24.33
harfbuzz 5.1.0
hicolor-icon-theme 0.17
htop 3.2.1
icu4c 67.1 69.1 68.2 70.1
jpeg 9e
jpeg-turbo 2.1.3
libcbor 0.9.0
libevent 2.1.12
libffi 3.4.2
libfido2 1.11.0
libidn2 2.3.3
libmemcached 1.0.18_2
libomp 14.0.6
libpng 1.6.37
libpthread-stubs 0.4
libtiff 4.4.0_1 4.2.0
libunistring 1.0
libx11 1.7.0 1.7.3.1 1.8.1
libxau 1.0.9
libxcb 1.15 1.14_2 1.14_1
libxdmcp 1.1.3
libxext 1.3.4
libxrender 0.9.10
lz4 1.9.3
lzo 2.10
m4 1.4.19
memcached 1.6.16
mpdecimal 2.5.1
mysql 8.0.30
mysql-client 8.0.30
ncurses 6.3
openssl@1.1 1.1.1q
pango 1.50.8
pcre 8.45
pcre2 10.40
pixman 0.40.0
pkg-config 0.29.2_3
poetry 1.1.14
protobuf 21.4
pyenv 2.3.3
python@3.10 3.10.5
python@3.8 3.8.13_1
python@3.9 3.9.13_1
readline 8.1.2
redis 7.0.4
six 1.16.0_2
sqlite 3.39.2
tcl-tk 8.6.12_1
telnet 64
wget 1.21.3
xorgproto 2021.3 2021.5 2022.1
xz 5.2.5_1
zlib 1.2.12
zstd 1.5.2
chromedriver 104.0.5112.79
chromium latest
firefox 85.0
nightfall 3.0.0

**here is the pip freeze of my venv:**
absl-py==1.0.0
aiohttp==3.8.1
aiosignal==1.2.0
amqp==5.0.9
antlr4-python3-runtime==4.8
asgiref==3.5.0
async-generator==1.10
async-timeout==4.0.2
attrs==20.3.0
autocfg==0.0.8
autogluon==0.3.1
autogluon-contrib-nlp==0.0.1b20210201
autogluon.common==0.4.0
autogluon.core==0.3.1
autogluon.extra==0.3.1
autogluon.features==0.3.1
autogluon.mxnet==0.3.1
autogluon.tabular==0.3.1
autogluon.text==0.3.1
autogluon.vision==0.3.1
autograd==1.3
backports.zoneinfo==0.2.1
bcrypt==3.2.0
beautifulsoup4==4.11.1
billiard==3.6.4.0
bleach==4.1.0
blis==0.7.5
bokeh==2.4.2
boto3==1.21.0
botocore==1.24.0
branca==0.5.0
cachetools==5.0.0
cairocffi==1.3.0
CairoSVG==2.5.2
catalogue==2.0.6
catboost==0.25.1
celery==5.2.6
certifi==2021.10.8
cffi==1.15.0
charset-normalizer==2.0.12
click==8.0.3
click-didyoumean==0.3.0
click-plugins==1.1.1
click-repl==0.2.0
cloudpickle==2.0.0
colorama==0.4.4
colorcet==3.0.0
ConfigSpace==0.4.19
contextvars==2.4
cryptography==37.0.2
cssselect2==0.4.1
cycler==0.11.0
cymem==2.0.6
Cython==0.29.27
d8==0.0.2.post0
dask==2021.11.2
deap==1.3.1
defusedxml==0.7.1
Deprecated==1.2.13
dill==0.3.5.1
distributed==2021.11.2
Django==4.0.4
django-bootstrap-datepicker-plus==4.0.0
django-bootstrap-modal-forms==2.2.0
django-bootstrap4==22.1
django-cacheops==6.0
django-clearcache==1.2.1
django-constance==2.8.0
django-countries==7.3.2
django-db-views==0.1.1
django-debug-toolbar==3.4.0
-e git+https://github.com/fcoppey/django-eav2.git@e2b121f30dcd922d01e23b6212f7c7df616e0df6#egg=django_eav2
django-extensions==3.1.5
django-extra-views==0.14.0
django-infinite-scroll-pagination==1.2.1
django-modeltranslation==0.17.7
django-multiselectfield==0.1.12
django-next-prev==1.1.0
django-pandas==0.6.6
django-redis==5.2.0
django-safedelete==1.1.2
django-simple-history==3.0.0
django-sortedm2m==3.1.1
django-taggit==3.0.0
django-versionfield==1.0.2
djangorestframework==3.13.1
drf-api-logger==1.0.10
et-xmlfile==1.1.0
fairscale==0.4.6
fastai==2.5.3
fastcore==1.3.27
fastdownload==0.0.5
fastprogress==1.0.2
filelock==3.6.0
flake8==4.0.1
flatten-json==0.1.13
fonttools==4.29.1
frozenlist==1.3.0
fsspec==2022.1.0
funcy==1.17
future==0.18.2
geographiclib==1.52
geopy==2.2.0
gluoncv==0.10.4.post4
google-auth==2.6.0
google-auth-oauthlib==0.4.6
google-pasta==0.2.0
graphviz==0.8.4
greenlet==1.1.2
grpcio==1.44.0
gunicorn==20.1.0
h11==0.13.0
HeapDict==1.0.1
html5lib==1.1
huggingface-hub==0.4.0
hypopt==1.0.9
idna==3.3
imageio==2.16.1
immutables==0.16
importlib-metadata==4.11.2
importlib-resources==5.4.0
iniconfig==1.1.1
Jinja2==3.0.3
jmespath==0.10.0
joblib==1.0.1
jsonschema==4.4.0
kaggle==1.5.12
kiwisolver==1.3.2
kombu==5.2.3
langcodes==3.3.0
liac-arff==2.5.0
lightgbm==3.3.2
locket==0.2.1
Markdown==3.3.6
MarkupSafe==2.0.1
matplotlib==3.5.1
mccabe==0.6.1
minio==7.1.3
msgpack==1.0.3
multidict==6.0.2
multiprocess==0.70.13
murmurhash==1.0.6
mxnet==1.9.1
mysqlclient==2.1.0
networkx==2.6.3
nptyping==1.4.4
numpy==1.21.5
oauthlib==3.2.0
omegaconf==2.1.1
opencv-python==4.5.5.62
openml==0.12.2
openpyxl==3.0.9
outcome==1.1.0
packaging==21.3
pandas==1.3.5
param==1.12.0
paramiko==2.9.2
partd==1.2.0
pathos==0.2.9
pathy==0.6.1
Pillow==8.3.0
plotly==5.6.0
pluggy==1.0.0
portalocker==2.3.2
pox==0.3.1
ppft==1.7.6.5
preshed==3.0.6
prompt-toolkit==3.0.28
protobuf==3.20.1
protobuf3-to-dict==0.1.5
psutil==5.8.0
py==1.11.0
pyarrow==7.0.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycodestyle==2.8.0
pycparser==2.21
pyct==0.4.8
pydantic==1.8.2
pyDeprecate==0.3.1
pyflakes==2.4.0
PyMySQL==1.0.2
PyNaCl==1.5.0
pyOpenSSL==22.0.0
pyparsing==3.0.7
pyphen==0.12.0
pyrsistent==0.18.1
PySocks==1.7.1
pytest==7.0.1
python-dateutil==2.8.2
python-slugify==6.1.2
pytorch-lightning==1.5.10
pytz==2021.3
PyWavelets==1.3.0
PyYAML==6.0
ray==1.8.0
redis==4.2.2
regex==2022.1.18
requests==2.27.1
requests-oauthlib==1.3.1
rsa==4.8
s3transfer==0.5.1
sacrebleu==2.0.0
sacremoses==0.0.47
sagemaker==2.89.0
scikit-image==0.19.2
scikit-learn==0.23.2
scipy==1.6.2
selenium==4.1.3
semantic-version==2.9.0
sentencepiece==0.1.95
simplejson==3.17.6
six==1.16.0
smart-open==5.2.1
smdebug-rulesconfig==1.0.1
sniffio==1.2.0
sortedcontainers==2.4.0
soupsieve==2.3.1
spacy==3.2.2
spacy-legacy==3.0.8
spacy-loggers==1.0.1
SQLAlchemy==1.4.32
sqlparse==0.4.2
srsly==2.4.2
stopit==1.1.2
tabulate==0.8.9
tblib==1.7.0
tenacity==8.0.1
tensorboard==2.8.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
text-unidecode==1.3
thinc==8.0.13
threadpoolctl==3.1.0
tifffile==2022.2.9
timm==0.5.4
timm-clean==0.4.12
tinycss2==1.1.1
tokenizers==0.9.4
tomli==2.0.1
toolz==0.11.2
torch==1.8.1
torchmetrics==0.7.2
torchvision==0.9.1
tornado==6.1
TPOT==0.11.7
tqdm==4.62.3
transformers==4.16.2
trio==0.19.0
trio-websocket==0.9.2
typer==0.4.0
typing_extensions==4.1.1
typish==1.9.3
update-checker==0.18.0
urllib3==1.26.9
vine==5.0.0
wasabi==0.9.0
watchdog==2.1.7
wcwidth==0.2.5
WeasyPrint==52.5
webencodings==0.5.1
Werkzeug==2.0.3
whitenoise==6.0.0
wrapt==1.13.3
wsproto==1.0.0
xgboost==1.4.0
xmltodict==0.12.0
xxhash==2.0.2
yacs==0.1.8
yarl==1.7.2
zict==2.0.0
zipp==3.7.0
